helix-core/src/main/java/org/apache/helix/model/management/ClusterManagementMode.java

```diff
@@ -0,0 +1,70 @@
+package org.apache.helix.model.management;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Represents the management mode of the cluster:
+ * 1. what type of mode it targets to be;
+ * 2. what progress status it is now.
+ */
+public class ClusterManagementMode {
```

Comment:
You can just make this class enum and list all the possible mode,
NORMAL()
CLUSTER_PAUSE(Status status) // inprogress or completed
CONTROLLER_PAUSE()
MAINTENANCE()

Then you don't need to face a Maintenance mode object with inprogress status. Which does not make sense.

Reply:
As discussed, this is not doable with enum. Let's keep the current design.